File: PB - More Exercises JavaScript/5.0 For-Loop - More Exercises/04. Grades.js
function grades(arg) {
    let students = Number(arg.shift())
    let grades = [...arg.map(parseFloat)]
    console.log(`Top students: ${(grades.filter(el => el >= 5 ).length * 100 / students).toFixed(2)}%`)
    console.log(`Between 4.00 and 4.99: ${(grades.filter(el => el >= 4  && el <= 4.99 ).length * 100 / students).toFixed(2)}%`)
    console.log(`Between 3.00 and 3.99: ${(grades.filter(el => el >= 3 && el <= 3.99 ).length * 100 / students).toFixed(2)}%`)
    console.log(`Fail: ${(grades.filter(el => el <= 2.99).length * 100 / students).toFixed(2)}%`)
    console.log(`Average: ${(grades.filter(el => el < 7).reduce((a, b) => a + b) / students).toFixed(2)}`)   
}

// grades([10, 3.00, 2.99, 5.68, 3.01, 4, 4, 6.00, 4.50, 2.44, 5, ])
// grades([6, 2, 3, 4, 5, 6, 2.2])

File: PB - More Exercises Python/5.0 For-Loop - More Exercises/04. Grades.py
students_count = int(input())
grades = [float(input()) for _ in range(students_count)]

def calc_grades(*args):
    test = 0
    if args[0] == 5:
        test = lambda x: x >= args[0]
    elif args[0] == 4:
        test = lambda x: args[0] <= x <= args[1]
    elif args[0] == 3:
        test = lambda x: args[0] <= x <= args[1]
    elif args[0] < 3:
        test = lambda x: args[0] > x

    return f'{len(list(filter(test, grades))) / students_count * 100:.2f}%'


print(f'Top students: {calc_grades(5)}')
print(f'Between 4.00 and 4.99: {calc_grades(4, 4.99)}')
print(f'Between 3.00 and 3.99: {calc_grades(3, 3.99)}')
print(f'Fail: {calc_grades(2.99)}')
print(f'Average: {sum(grades) / students_count:.2f}')


'''
6
2
3
4
5
6
2.2
'''
